internal/gomod/version.go

```diff
@@ -69,99 +69,117 @@ func GetModuleVersion(moduleDir string) (string, error) {
 // GetVersionFromTag checks if the HEAD commit is annotated with a tag and if it is, returns that tag's name.
 // If the HEAD commit is not tagged, a pseudo version will be generated and returned instead.
 func GetVersionFromTag(moduleDir string) (string, error) {
-	repo, err := git.PlainOpen(moduleDir)
+	headCommit, err := doExec("sh", []string{"-c", "git show-ref --heads | cut -d' ' -f1"}, moduleDir, nil)
 	if err != nil {
 		return "", err
 	}
 
-	headRef, err := repo.Head()
+	latestTag, err := doExec("sh", []string{"-c", "git describe --tags $(git rev-list --tags --max-count=1)"}, moduleDir, nil)
 	if err != nil {
 		return "", err
 	}
 
-	headCommit, err := repo.CommitObject(headRef.Hash())
+	latestTagCommit, err := doExec("sh", []string{"-c", fmt.Sprintf("git show-ref -s %s", string(latestTag))}, moduleDir, nil)
```

Comment:
I think we can replace these with `git describe --tags --abbrev=0 <COMMIT_SHA>`. This will give us the tags that point to `<COMMIT_SHA>` (which is `HEAD` in our case).  

`git describe --tags $(git rev-list --tags --max-count=1)` will yield the latest tag *across all branches*, which may not be the tag we generated the SBOM for.